modules/system/classes/PluginManager.php

```diff
@@ -855,6 +855,9 @@ public function deletePlugin($id)
         if ($pluginPath = self::instance()->getPluginPath($id)) {
             File::deleteDirectory($pluginPath);
         }
+
+        // actually remove the plugin from our internal container
+        unset($this->plugins[$id]);
```

Comment:
Is the ID correctly normalized at this point?

Reply:
not necessarily, because it can be called from outside the class... I'll add normalization to make sure.